app/javascript/src/javascripts/posts.js

```diff
@@ -354,9 +354,9 @@ Post.initialize_links = function() {
   });
   $("#destroy-post-link").on('click', e => {
     e.preventDefault();
-    if(!confirm("This will permanently delete this post (meaning the file will be deleted). Are you sure you want to delete this post?"))
-      return;
-    Post.destroy($(e.target).data('pid'));
+    const reason = prompt("This will permanently delete this post (meaning the file will be deleted). What is the reason for destroying the post?")
+    if(reason === null) return;
```

Comment:
Reason would never be null here.
If the user leaves the prompt empty, it'll simply return an empty string.

Reply:
Canceling returns null, if we don't check for null it's impossible to cancel

Reply:
Right, okay.